### PR TITLE
Add Loan returned handler Lambda

### PR DESCRIPTION
**Pull request by DomBelcher**
This adds a Lambda for handling SNS events of type LOAN_RETURNED.

It deletes the corresponding Loan record from the cache, and removes the Loan ID from the matching User record

### CODE REVIEW

File: src/cache.js [hunk 1]
```diff
@@ -26,38 +26,61 @@ class Cache {
...
+          return {
+            'add': {
+              'loan': user.addLoan,
+              'request': user.addRequest
+            },
+            'delete': {
+              'loan': user.deleteLoan,
+              'request': user.deleteRequest
+            }
+          }[operation][itemType].call(user, itemID).save()
```
Comment (stephenrob) [inline src/cache.js]: How does this handle operation or itemType being `null` or `undefined`

File: src/cache.js [hunk 2]
```diff
@@ -26,38 +26,61 @@ class Cache {
...
+      this.deleteLoan(itemLoan.loan_id),
+      this.updateUserWithDeleteLoan(itemLoan.user_id, itemLoan.loan_id)
+    ])
+  }
+
   handleLoanUpdate (itemLoan) {
     return Promise.all([
       this.updateLoan(itemLoan),
-      this.updateUserWithLoan(itemLoan.user_id, itemLoan.loan_id)
+      this.updateUserWithAddLoan(itemLoan.user_id, itemLoan.loan_id)
```
Comment (stephenrob) [inline src/cache.js]: Not sure this new naming of the method is clear or not. Maybe `updateUserAndAddLoan`, not overly happy with my name suggestion.

**Review by stephenrob — COMMENTED** [review]
Few comments regarding naming of methods and their behaviour.

File: src/cache.js (hunk 1 above)
Comment (DomBelcher) [reply src/cache.js]: It's not really a method that's meant to be called from outside the class, it's just Javascript doesn't have proper private class methods, so it's only meant to be called internally with valid arguments even though it is public. 

Could add some sort of handling, either checking the arguments are valid or just wrap it in a try/catch. It's also possible to make it "private" in a slightly hacky way using `call`, not sure which would be better though.

Comment (stephenrob) [reply src/cache.js]: I'd be happy with adding either a try/catch or some argument validation as its been called this way.

File: src/cache.js (hunk 2 above)
Comment (DomBelcher) [reply src/cache.js]: Possibly something along the lines of `addLoanToUser`, `deleteLoanFromUser` would be better?

Comment (stephenrob) [reply src/cache.js]: That naming is a lot clearer

File: src/loan-returned/handler.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const extractMessageData = require('../extract-message-data')
+const Cache = require('../cache')
+
+module.exports.handle = (event, context, callback) => {
+  let loanData
+
+  let almaCache = new Cache({
```
Comment (stephenrob) [inline src/loan-returned/handler.js]: Can this be extracted out external to this that when required returns an instance of the almaCache from the environment? Would help reduce duplication

**Review by stephenrob — APPROVED** [review]

**Pull request merged**